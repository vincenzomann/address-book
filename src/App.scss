* {
	margin: 0;
	padding: 0;
	box-sizing: border-box;
}

html,
body {
	height: 100%;
	width: 100%;
}

#root,
#app {
	height: 100%;
	width: 100%;
	display: flex;
	flex-direction: row;
	font-family: DM Sans, sans-serif;

	#sideNav {
		height: 100%;
		min-width: 72px;
		border-right: 1px solid lightGrey;
		text-align: center;
		@media only screen and (min-width: 700px) {
			min-width: 240px;
			text-align: left;
		}

		#logoItem {
			padding: 1.5rem 1rem;

			.logo {
				height: 2.25rem;
			}
			#logo-full {
				display: none;
				@media only screen and (min-width: 700px) {
					display: inline;
				}
			}
			#logo-mob {
				display: inline;
				@media only screen and (min-width: 700px) {
					display: none;
				}
			}
		}

		#navItems {
			:hover {
				background-color: #f5f5f5;
				cursor: pointer;
				* {
					color: #009bc1;
				}
			}

			.navItem {
				padding: 0.5rem 1rem;
				display: flex;
				align-items: center;
				.icon {
					color: grey;
					@media only screen and (min-width: 700px) {
						padding: 0 0.25rem;
					}
				}
				.linkText {
					text-decoration: none;
					font-size: small;
					display: none;
					padding: 1rem;
					@media only screen and (min-width: 700px) {
						display: inline;
					}
				}
			}
		}
	}

	#content {
		height: 100%;
		width: 100%;
		background-color: #f5f5f5;
		padding: 1rem;

		#lookup {
			padding: 1rem;
			#pageTitle {
				font-weight: bold;
			}

			#lookupContainer {
				display: flex;
				justify-content: center;
				box-sizing: border-box;
				max-width: 800px;
				margin: 2rem auto;

				#switch {
					:hover {
						cursor: pointer;
					}
					.option {
						margin-right: 1rem;
						font-size: large;
					}
					.selected {
						border-bottom: 3px solid #009bc1;
						padding-bottom: 0.25rem;
					}
				}

				#form {
					display: flex;
					justify-content: center;
					flex-direction: column;
					.inputs {
						padding: 1rem 0;
						input {
							height: 2.5rem;
							width: 100%;
							margin: 0.5rem 0;
							border: 1px solid lightGrey;
							border-radius: 10px;
							padding: 1rem;
						}
					}

					#select {
						margin: 0.5rem 0;
						font-size: small;

						input {
							height: 100%;
						}
					}

					.error {
						height: 48px;
						width: 100%;
						font-size: 18px;
						color: #f36f5e;
						margin: 0 1rem;
					}

					#searchBtn {
						height: 48px;
						width: 100%;
						font-size: 18px;
						color: #009bc1;
						background: #e5f5f8;
						font-weight: bold;
						border: none;
						border-radius: 10px;
						display: flex;
						justify-content: center;
						align-items: center;
					}
				}
			}
		}
	}
}

.card {
	width: 100%;
	background-color: white;
	border: 1px solid lightGrey;
	border-radius: 10px;
	padding: 2rem;
	box-shadow: 0 2px 10px rgb(0 0 0 / 10%);
}
