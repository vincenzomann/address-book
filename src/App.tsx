import React, { useEffect } from 'react';
import './App.scss';
import { navigate, Router } from '@reach/router';
import AddressBook from './components/AddressBook';
import Lookup from './components/Lookup';
import ContextProvider from './context/Context';
import SideNav from './components/SideNav';

function App() {

	useEffect(() => {
		if (window.location.pathname === '/') {
			navigate('/address-book');
		}
	}, []);

	return (
		<div id="app">
			<ContextProvider >
				<SideNav />
				<div id="content">
					<Router>
						<AddressBook path='/address-book' default />
						<Lookup path='/lookup' />
					</Router>
				</div>
			</ContextProvider>
		</div>
	);
}

export default App;
