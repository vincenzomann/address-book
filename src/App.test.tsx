import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

test('renders the component', () => {
	const wrapper = shallow(<App />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
