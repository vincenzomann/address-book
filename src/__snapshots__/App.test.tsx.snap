// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="app"
  id="app"
>
  <ContextProvider>
    <SideNav />
    <div
      id="content"
    >
      <Router>
        <AddressBook
          default={true}
          path="/address-book"
        />
        <Lookup
          path="/add-address"
        />
      </Router>
    </div>
  </ContextProvider>
</div>
`;

exports[`renders the component 1`] = `
<div
  data-testid="app"
  id="app"
>
  <ContextProvider>
    <SideNav />
    <div
      id="content"
    >
      <Router>
        <AddressBook
          default={true}
          path="/address-book"
        />
        <Lookup
          path="/add-address"
        />
      </Router>
    </div>
  </ContextProvider>
</div>
`;
